blob/src/main/java/com/bazaarvoice/emodb/blob/BlobStoreModule.java

```diff
@@ -385,32 +394,62 @@ ConsistencyLevel provideBlobReadConsistency(BlobStoreConfiguration configuration
     }
 
     @Provides @Singleton @S3BucketNamesToS3Clients
-    Map<String, AmazonS3> provideS3BucketNamesToS3Clients(BlobStoreConfiguration configuration) {
+    Map<String, AmazonS3> provideS3BucketNamesToS3Clients(BlobStoreConfiguration configuration, Clock clock) {
+        S3RateLimiter rateLimiter = new S3RateLimiter(clock);
         S3Configuration s3Configuration = configuration.getS3Configuration();
         //    TODO remove condition in EMO-7107
         if (null != s3Configuration && null != s3Configuration.getS3BucketConfigurations()) {
             return s3Configuration.getS3BucketConfigurations().stream()
                     .collect(Collectors.toMap(
                             s3BucketConfiguration -> s3BucketConfiguration.getName(),
-                            s3BucketConfiguration -> {
-                                AmazonS3ClientBuilder amazonS3ClientBuilder = AmazonS3ClientBuilder.standard()
-                                        .withRegion(Regions.fromName(s3BucketConfiguration.getRegion()))
-                                        .withCredentials(getAwsCredentialsProvider(s3BucketConfiguration))
-                                        .withAccelerateModeEnabled(s3BucketConfiguration.getAccelerateModeEnabled());
-                                if (null != s3Configuration.getS3ClientConfiguration()) {
-                                    S3ClientConfiguration.EndpointConfiguration endpointConfiguration = s3Configuration.getS3ClientConfiguration().getEndpointConfiguration();
-                                    amazonS3ClientBuilder
-                                            .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endpointConfiguration.getServiceEndpoint(), endpointConfiguration.getSigningRegion()));
-                                }
-                                return amazonS3ClientBuilder
-                                        .build();
-                            })
+                            s3BucketConfiguration -> rateLimiter.rateLimit(getAmazonS3(s3BucketConfiguration)))
                     );
         } else {
             return new HashMap<>();
         }
     }
 
+    private static AmazonS3 getAmazonS3(S3BucketConfiguration s3BucketConfiguration) {
```

Comment:
Considering the extreme complexity of the client configuration, can we extract this from the Guice module, and relocate into a separate, self-contained class, similar to `PlacementFactory`?

Reply:
would leave it in MVP and create ticket to refactor (if that's really needed).